lib/midcom/helper/imagepopup/handler/upload.php

```diff
@@ -0,0 +1,170 @@
+            {
+                midcom::get ()->header ( "HTTP/1.0 500 Invalid file name." );
+                return;
+            }
+            
+            if (! in_array ( strtolower ( pathinfo ( $temp ['name'], PATHINFO_EXTENSION ) ), array (
+                            "gif",
+                            "jpg",
+                            "png" 
+            ) ))
+            {
+                midcom::get ()->header ( "HTTP/1.0 500 Invalid extension." );
+                return;
+            }
+        }
+        
+        // Get the data
+        $temp_name = $temp ['tmp_name'];
+        $mimetype = $temp ['type'];
+        $parentguid = "";
+        
+        // Do we have guid ?
+        switch($handler_id)
+        {
+            case '____ais-imagepopup-upload_image_noobject':
+                $parentguid = $this->_topic->guid;
+                break;
+            case '____ais-imagepopup-upload_image':
+                $parentguid = $args[0];
+                break;
+        }
+        
+        // Set modified filename
+        $filename = $this->get_modify_filename ( $temp ['name'] );
+        
+        // Insert the image into database
+        $attachment = $this->insert_database ( $filename, $mimetype, $parentguid );
+        
+        // Get target to write the file
+        $target = $this->get_data_from_database ( $filename, $parentguid );
+        
+        // Write the file
+        $this->write_the_file ( $temp_name, $target );
+        
+        // Make a response for editor.uploadImages() function
+        $location = midcom_db_attachment::get_url ( $attachment );
+        
+        // Return image location as JSON
+        return new midcom_response_json ( array (
+                        'location' => $location 
+        ) );
+    }
+    
+    private function get_modify_filename($filename)
+    {
+        $file_name = midcom_db_attachment::safe_filename ( $filename );
+        $pieces = explode ( '.', $file_name );
+        $core = "modified";
+        $extension = array_pop ( $pieces );
+        $time = $this->get_current_time ( 'd-m-Y-H-i-s' );
+        $modifyFilename = $core . $time . "." . $extension;
+        
+        return $modifyFilename;
+    }
+    
+    private function get_current_time($pattern)
+    {
+        return date ( $pattern, time () );
+    }
+    
+    private function get_data_from_database($filename, $parentguid)
+    {
+        $file_name = midcom_db_attachment::safe_filename ( $filename );
+        $query = midcom_db_attachment::new_query_builder ();
+        $query->add_constraint ( 'name', '=', $file_name );
+        $query->add_constraint ( 'parentguid', '=', $parentguid );
+        $entry = $query->execute ();
+        
+        if (count ( $entry ) == 1)
+        {
+            return $entry [0];
+        } else if (count ( $entry ) > 1)
+        {
+            throw new midcom_error ( 'There is more than just one object' . midcom_connection::get_error_string () );
+        } else
```

Comment:
if the previous blocks aborts (by returning a value or throwing an exception) you don't really need to use `else`